var/spack/repos/builtin/packages/libogg/package.py

```diff
@@ -22,6 +23,14 @@ class Libogg(CMakePackage, AutotoolsPackage, Package):
     version("1.3.4", sha256="fe5670640bd49e828d64d2879c31cb4dde9758681bb664f9bdbf159a01b0c76e")
     version("1.3.2", sha256="e19ee34711d7af328cb26287f4137e70630e7261b17cbe3cd41011d73a654692")
 
+    variant("shared", default=True, description="Build shared library", when="build_system=cmake")
```

Comment:
FYI some pkgs have this be dependent on `+pic` (adios2, w3emc).
```py
    variant("shared", default=True, when="+pic", description="Build shared libraries")
```
Does that make sense here too?

Reply:
In this case, shared needs to be a variant that then goes on to trigger `+pic`. However, triggering `+shared` from `+pic` also makes sense. My spack-foo isn't good enough, but can there be a conditional of `when="+shared or +pic"`?

Reply:
@wdconinc thinking further: CMake guarantees that PIC is set if shared is requested, so in this case the behaviour is as expected. I think it's fine to leave it unless @haampie has further thoughts (as I bugged him in DM for spackthonic way).

Reply:
I have added a requires on `+pic` when `+shared` just to be absolutely sure `+shared~pic` can't trick cmake